workflow/snakemake_rules/nextclade.smk

```diff
@@ -162,18 +162,26 @@ rule combine_alignments:
         fi
         """
 
-def _get_nextclade_info(wildcards):
+def _get_nextclade_output(wildcards):
     ## the nextclade metadata should represent the entire dataset. If there are new sequences
     ## this has to be generated; if not then we can use the previous (cached) file.
     nextclade_sequences_path = checkpoints.get_sequences_without_nextclade_annotations.get().output.fasta
     if os.path.getsize(nextclade_sequences_path) > 0:
-        return f"data/{database}/nextclade.tsv"
-    return f"data/{database}/nextclade_old.tsv"
+        # Return all three output files here so that they get pulled through
+        # the Snakemake DAG even if we are not uploading them to S3.
+        return [
+            f"data/{database}/nextclade.tsv",
+            f"data/{database}/aligned.fasta",
+        ]
+    return [
+        f"data/{database}/nextclade_old.tsv",
+        f"data/{database}/nextclade.aligned.old.fasta",
+    ]
 
 rule generate_metadata:
     input:
         existing_metadata = f"data/{database}/metadata_transformed.tsv",
-        new_metadata = _get_nextclade_info
+        nextclade_output = _get_nextclade_output
```

Comment:
If `_get_nextclade_output` returned a dictionary instead of a list, you could do:

```
input:
    unpack(_get_nextclade_output),
    existing_metadata = f"data/{database}/metadata_transformed.tsv"
```

to make the inputs individually keyed. See https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#input-functions-and-unpack.

This would help with the shell block for this rule… (see below)

Reply:
Ah yup, just recently learned by about the `unpack` keyword. I'll update this.

Reply:
Fixed up in 30f0e6a6fd94c24996377a7434671605dfb0a2e2. I renamed `input.new_metadata` to `input.nextclade_tsv` to make it clearer that we are merging metadata with NextClade data.